src/YUIException.h

```diff
@@ -172,11 +172,27 @@ class YWidget;
 /**
  * Check if a widget pointer is valid.
  * Throws YUIInvalidWidgetException if it is 0 or invalid (already deleted).
+ *
+ * Explicitly casting the memory-address stored in the given pointer to
+ * a boolean-type for null-pointer-checks is needed for GCC >= 6, because
+ * it introduces new optimizations to remove null-pointer-checks for 'this'.
+ *
+ * Not explicitly casting the pointer's memory-address, will cause the
+ * compilation to fail with an error, when using this macro in YDialog:
```

Comment:
I guess, gcc6 actually throws a warning and compilation only breaks because libyui is compiled with -Werror which often breaks things with new compilers that introduce new warnings.